gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/SettingsGradingSchemaPanel.java

```diff
@@ -138,7 +138,12 @@ public void onInitialize() {
 		// create map of grading scales to use for the dropdown
 		final Map<String, String> gradeMappingMap = new LinkedHashMap<>();
 		for (final GradeMappingDefinition gradeMapping : this.gradeMappings) {
-			gradeMappingMap.put(gradeMapping.getId(), new ResourceModel("settingspage.gradingschema.gradetypes." + gradeMapping.getName()).getObject());
+			String value = new ResourceModel("settingspage.gradingschema.gradetypes." + gradeMapping.getName()).getObject();
+			if (!value.contains("missing key")) {
+				gradeMappingMap.put(gradeMapping.getId(), new ResourceModel("settingspage.gradingschema.gradetypes." + gradeMapping.getName()).getObject());
```

Comment:
You've already obtained the `String` from the `ResourceModel` and confirmed it doesn't contain "missing key", so you don't need to create another new `ResourceModel` on line 143, just pass in the `String` you already retrieved:

```suggestion
			String value = new ResourceModel("settingspage.gradingschema.gradetypes." + gradeMapping.getName()).getObject();
			if (!value.contains("missing key")) {
				gradeMappingMap.put(gradeMapping.getId(), value);
```

Reply:
Thanks @bjones86! Too much work this morning...